src/baseline.rs

```diff
@@ -338,21 +456,7 @@ impl BaselineLoader for RegistryBaseline {
             choose_baseline_version(&crate_, version_current)?
         };
 
-        let crate_identifier = format!("registry-{}-{}", slugify(name), slugify(&base_version));
-
-        let cache_dir = self.target_root.join("cache");
-        let cached_rustdoc = cache_dir.join(format!("{crate_identifier}.json"));
-
-        // We assume that the generated rustdoc is untouched. Users should run cargo-clean if they experience any anomalies.
-        if cached_rustdoc.exists() {
-            return Ok(cached_rustdoc);
-        }
```

Comment:
@tonowak I don't think you ported this bit correctly. The old logic here is correct but the new logic added in this PR has several different bugs.